.github/workflows/setup_env.sh

```diff
@@ -0,0 +1,141 @@
+#!/bin/bash
+
+#-------------------------------------------------------------------------------
+# Functions
+
+# Suppress echo (-x) output of commands executed with `quiet`.
+# Useful for sourcing files, loading modules, spack, etc.
+# set +x, set -x are not echo'd.
+quiet() {
+    { set +x; } 2> /dev/null;
+    $@;
+    set -x
+}
+
+# `print` is like `echo`, but suppresses output of the command itself.
+# https://superuser.com/a/1141026
+echo_and_restore() {
+    builtin echo "$*"
+    date
+    case "${save_flags}" in
+        (*x*)  set -x
+    esac
+}
+alias print='{ save_flags="$-"; set +x; } 2> /dev/null; echo_and_restore'
+
+
+#-------------------------------------------------------------------------------
+quiet source /etc/profile
+
+hostname && pwd
+export top=$(pwd)
```

Comment:
I see the export here. Should we export it as full capital letters?

Reply:
Generally across all our code I've used the same style, which is lowercase variables unless there's a compelling reason like other apps spell it differently (e.g., $CXX). I have a habit of exporting all variables; maybe I shouldn't.